app/code/Magento/InventoryConfiguration/Model/StockItemConditionInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfiguration\Model;
+
+/**
+ * Interface StockConditionInterface
+ * @package Magento\InventoryConfiguration\Model
+ */
+interface StockItemConditionInterface
```

Comment:
So... 
what you've implemented is additional service which is passed to isProductSalable instance. 

And what isProductSalable does is call all the Condition services 
```
 foreach ($this->conditions as $condition) {
        if (null === $stockItemData) {	            
                 if ($condition->match($sku, $stockId) === true) {
                return false;	
                return true;
       }	
```
isProductSalable looks and behaves like `Composite` object. So, the quetion why do we need to have 2 composites: isProductSalable and StockItemCondition? 

What we need to have - the only composite with a clean and understandable interface. Moreover we already have this interface. Its name is `isProductSalableInterface`

Thus, what we need to do - is remove `StockItemConditionInterface` and make all its implementations to implement `isProductSalableInterface` interface.
We will end up having 1 composite object isProductSalable which will accept an array of isProductSalableInterface[] and do similar operation 
```
 foreach ($this->services as $isProductSalable) {    
                 if ($isProductSalable->execute($sku, $stockId) === true) {
                     return true;	
      //...
       }	
```

Reply:
Done